yt_dlp/extractor/microsoftembed.py

```diff
@@ -63,3 +71,253 @@ def _real_extract(self, url):
             'thumbnails': thumbnails,
         }
+
+
+class MicrosoftMediusBaseIE(InfoExtractor):
+    @staticmethod
+    def _sub_to_dict(subtitle_list):
+        subtitles = {}
+        for sub in subtitle_list:
+            subtitles.setdefault(sub.pop('tag', 'unknown'), []).append(sub)
+        return subtitles
+
+    def _extract_ism(self, ism_url, video_id):
+        formats = self._extract_ism_formats(ism_url, video_id)
+        for fmt in formats:
+            if fmt.get('language') == 'eng' or 'English' in fmt.get('format_id', ''):
+                fmt['language_preference'] = -1
+            else:
+                fmt['language_preference'] = -10
+        return formats
+
+
+class MicrosoftMediusIE(MicrosoftMediusBaseIE):
+    _VALID_URL = r'https?://medius\.microsoft\.com/Embed/(?:Video\?id=|video-nc/|VideoDetails/)(?P<id>[0-9a-f\-]+)'
+
+    _TESTS = [{
+        'url': 'https://medius.microsoft.com/Embed/video-nc/9640d86c-f513-4889-959e-5dace86e7d2b',
+        'info_dict': {
+            'id': '9640d86c-f513-4889-959e-5dace86e7d2b',
+            'ext': 'ismv',
+            'title': 'Rapidly code, test and ship from secure cloud developer environments',
+            'description': 'md5:33c8e4facadc438613476eea24165f71',
+            'thumbnail': r're:https://mediusimg\.event\.microsoft\.com/video-\d+/thumbnail\.jpg.*',
+            'subtitles': 'count:30',
+        },
+    }, {
+        'url': 'https://medius.microsoft.com/Embed/video-nc/81215af5-c813-4dcd-aede-94f4e1a7daa3',
+        'info_dict': {
+            'id': '81215af5-c813-4dcd-aede-94f4e1a7daa3',
+            'ext': 'ismv',
+            'title': 'Microsoft Build opening',
+            'description': 'md5:43455096141077a1f23144cab8cec1cb',
+            'thumbnail': r're:https://mediusimg\.event\.microsoft\.com/video-\d+/thumbnail\.jpg.*',
+            'subtitles': 'count:31',
+        },
+    }, {
+        'url': 'https://medius.microsoft.com/Embed/VideoDetails/78493569-9b3b-4a85-a409-ee76e789e25c',
+        'info_dict': {
+            'id': '78493569-9b3b-4a85-a409-ee76e789e25c',
+            'ext': 'ismv',
+            'title': ' Anomaly Detection & Root cause at Edge',
+            'description': 'md5:f8f1ad93d7918649bfb97fa081b03b83',
+            'thumbnail': r're:https://mediusdownload.event.microsoft.com/asset.*\.jpg.*',
+            'subtitles': 'count:17',
+        },
+    }, {
+        'url': 'https://medius.microsoft.com/Embed/Video?id=0dc69bda-079b-4070-a7db-a8da1a06a9c7',
+        'only_matching': True,
+    }, {
+        'url': 'https://medius.microsoft.com/Embed/video-nc/fe823a91-959c-465b-96d4-8f4db624f72c',
+        'only_matching': True,
+    }]
+
+    def _extract_subtitle(self, webpage, video_id):
+        captions = traverse_obj(
+            self._search_json(r'const\s+captionsConfiguration\s*=', webpage, 'captions', video_id, default=None),
+            ('languageList', ..., {
+                'url': ('src', {url_or_none}),
```

Comment:
instead of transforming with `url_or_none`, I would filter with it so that we don't return `'url': None`
```suggestion
            ('languageList', lambda _, v: url_or_none(v['src']), {
                'url': 'src',
```